roles/v2/pool/src/lib/mining_pool/mod.rs

```diff
@@ -273,36 +317,55 @@ impl Pool {
         rx: Receiver<NewTemplate<'static>>,
         sender_message_received_signal: Sender<()>,
     ) {
+        let status_tx = self_.safe_lock(|s| s.status_tx.clone()).unwrap();
         while let Ok(mut new_template) = rx.recv().await {
             debug!(
                 "New template received, creating a new mining job(s): {:?}",
                 new_template
             );
 
-            let channel_factory = self_.safe_lock(|s| s.channel_factory.clone()).unwrap();
-            let mut messages = channel_factory
-                .safe_lock(|cf| cf.on_new_template(&mut new_template).unwrap())
-                .unwrap();
-            let downstreams = self_.safe_lock(|s| s.downstreams.clone()).unwrap();
+            let channel_factory = self_
```

Comment:
Do you think we need to clone the channel_factory on every  message - or can this be outside the loop?